datetimeparser/enums.py

```diff
@@ -1,41 +1,44 @@
-from .baseclasses import Constant, Constant, MethodEnum
+from datetime import datetime, timedelta
+
+from .baseclasses import Constant, MethodEnum
+from .formulars import days_feb, eastern_calc, thanksgiving_calc, year_start
 
 
 class Constants:
-    CHRISTMAS = Constant('christmas', ['xmas'])
-    SILVESTER = Constant('silvester', ['new years eve'])
-    EASTERN = Constant('eastern', ['easter'])
-    NICHOLAS = Constant('nicholas', ['nicholas day'])
-    HALLOWEEN = Constant('halloween')
-    APRIL_FOOLS_DAY = Constant('april fools day', ['april fool day'])
-    THANKSGIVING = Constant('thanksgiving')
-    SAINT_PATRICKS_DAY = Constant('saint patrick\'s day', ['saint patricks day', 'st. patrick\'s day', 'saint pt. day', 'st patrick\'s day', 'st patricks day'])
-    VALENTINES_DAY = Constant('valentines day', ['valentine', 'valentine day'])
-
-    SUMMER_BEGIN = Constant('summer begin', ['summer', 'begin of summer', 'begin of the summer'])
-    WINTER_BEGIN = Constant('winter begin', ['winter', 'begin of winter', 'begin of the winter'])
-    SPRING_BEGIN = Constant('spring begin', ['spring', 'begin of spring', 'begin of the spring'])
+    CHRISTMAS = Constant('christmas', ['xmas'], time_value=lambda year_time: datetime(year=year_time, month=12, day=25))
+    SILVESTER = Constant('silvester', ['new years eve'], time_value=lambda year_time: datetime(year=year_time, month=12, day=31))
+    EASTERN = Constant('eastern', ['easter'], time_value=eastern_calc)
+    NICHOLAS = Constant('nicholas', ['nicholas day'], time_value=lambda year_time: datetime(year=year_time, month=12, day=6))
+    HALLOWEEN = Constant('halloween', time_value=lambda year_time: datetime(year=year_time, month=10, day=31))
+    APRIL_FOOLS_DAY = Constant('april fools day', ['april fool day'], time_value=lambda year_time: datetime(year=year_time, month=4, day=1))
+    THANKSGIVING = Constant('thanksgiving', time_value=thanksgiving_calc)
+    SAINT_PATRICKS_DAY = Constant('saint patrick\'s day', ['saint patricks day', 'st. patrick\'s day', 'saint pt. day', 'st patrick\'s day', 'st patricks day'], time_value=lambda year_time: datetime(year=year_time, month=3, day=17))
+    VALENTINES_DAY = Constant('valentines day', ['valentine', 'valentine day'], time_value=lambda year_time: datetime(year=year_time, month=2, day=14))
+
+    SUMMER_BEGIN = Constant('summer begin', ['summer', 'begin of summer', 'begin of the summer'], time_value=lambda year_time: datetime(year=year_time, month=6, day=1))
+    WINTER_BEGIN = Constant('winter begin', ['winter', 'begin of winter', 'begin of the winter'], time_value=lambda year_time: datetime(year=year_time, month=12, day=1))
+    SPRING_BEGIN = Constant('spring begin', ['spring', 'begin of spring', 'begin of the spring'], time_value=lambda year_time: datetime(year=year_time, month=3, day=1))
```

Comment:
Line too long, look at other lines where they've been already splitted onto multiple lines and do the same